src/governance.cpp

```diff
@@ -954,23 +954,22 @@ void CGovernanceManager::RequestGovernanceObject(CNode* pfrom, const uint256& nH
         return;
     }
 
-    CBloomFilter filter(GOVERNANCE_FILTER_ELEMENTS, GOVERNANCE_FILTER_FP_RATE, GetRandInt(999999), BLOOM_UPDATE_ALL);
+    CBloomFilter filter;
+    filter.clear();
 
     if(fUseFilter) {
         CGovernanceObject* pObj = FindGovernanceObject(nHash);
 
         if(pObj) {
+            int nFilterElements = min(3*mnodeman.CountMasternodes(), GOVERNANCE_FILTER_MAX_ELEMENTS); 
```

Comment:
I doubt this would change anything for testnet... Would be nice to make it net-dependent imo i.e. define smth like `nGovernanceFilterMaxElements` (and corresponding `get` function) in `chainparams.h`, set it to say 15000 for mainnet, 500 for testnet, 100(?) regtest in `chainparams.cpp` and use it here instead of a constant.